hugegraph-core/src/main/java/com/baidu/hugegraph/backend/tx/GraphIndexTransaction.java

```diff
@@ -1102,13 +1103,33 @@ public void rebuildIndex(HugeType type, Id label,
              * They have different id lead to it can't compare and optimize
              */
             this.commit();
-            if (type == HugeType.VERTEX_LABEL) {
-                this.rebuildIndex(HugeType.VERTEX, label, indexLabelIds,
-                                  graphTx::queryVerticesFromBackend);
+            if (label.type() == HugeType.VERTEX_LABEL) {
+                graphTx.traverseVerticesByLabel(
+                        (VertexLabel) label,
+                        vertex -> {
```

Comment:
define Consumer<HugeElement> consumer
and let it be shared by traverseVerticesByLabel and traverseEdgesByLabel